tests/integration/test_orders_api.py

```diff
@@ -536,6 +536,222 @@ async def test_download_asset_md(tmpdir, session):
     assert Path(filename).name == 'metadata.json'
 
 
+@respx.mock
+@pytest.mark.asyncio
+async def test_checksum_success(tmpdir, order_description, oid, session):
+    # Mock an HTTP response for download
+    order_description['state'] = 'success'
+    dl_url1 = TEST_DOWNLOAD_URL + '/1?token=IAmAToken'
+    dl_url2 = TEST_DOWNLOAD_URL + '/2?token=IAmAnotherToken'
+    dl_url3 = TEST_DOWNLOAD_URL + '/2?token=IAmYetAnotherToken'
+    order_description['_links']['results'] = [{
+        'location': dl_url1
+    }, {
+        'location': dl_url2
+    }, {
+        'location': dl_url3
+    }]
+    get_url = f'{TEST_ORDERS_URL}/{oid}'
+    mock_asset = httpx.Response(HTTPStatus.OK, json=order_description)
+    respx.get(get_url).return_value = mock_asset
+
+    mock_asset1 = httpx.Response(HTTPStatus.OK,
+                                 json={
+                                     "md5":
+                                     "1975b76cc6904ff84f432be2ef1fea2e",
+                                     "sha256":
+                                     "300ebdfd62614f82ce6ba45b298bf797\
+                                           f1ea595d9214bcf23899f3fd2267c5dd"
+                                 },
+                                 headers={
+                                     'Content-Type':
+                                     'application/json',
+                                     'Content-Disposition':
+                                     'attachment; filename="asset1.json"'
+                                 })
+    respx.get(dl_url1).return_value = mock_asset1
+
+    mock_asset2 = httpx.Response(HTTPStatus.OK,
+                                 json={
+                                     "md5":
+                                     "27323e74b2590f97734124bd4cce055e",
+                                     "sha256":
+                                     "cc70916fe9af731d80220aec6dfbf7c\
+                                          91b9339fea61bf2a6b4d83cee6e11f7c9"
+                                 },
+                                 headers={
+                                     'Content-Type':
+                                     'application/json',
+                                     'Content-Disposition':
+                                     'attachment; filename="asset2.json"'
+                                 })
+    respx.get(dl_url2).return_value = mock_asset2
+
+    mock_manifest = httpx.Response(
+        HTTPStatus.OK,
+        json={
+            "name":
+            "",
+            "files": [{
+                "path": "asset1.json",
+                "digests": {
+                    "md5":
+                    "9a967dfe809c2eb25d990ad547feadf0",
```

Comment:
It would also be okay to compute the digest values directly in here. Like this:

```suggestion
                    hashlib.md5(mock_asset1.content).hexdigest(),
```

Then it would automatically stay up to date if we ever changed the mock response JSON.

BTW, I verified that we can use the `content` attribute of a response like that.

```pycon
>>> resp = Response(200, json={"asset": 1})
>>> resp
<Response [200 OK]>
>>> resp.content
b'{"asset": 1}'
```

It's always bytes, so no need to encode it.

Reply:
This could also help make flake8 happy.